### PR TITLE
Porting to cpp data

### PR DESCRIPTION
**Pull request by wdigger**
Data loading and access ported to C++.
I made some changes:
- sprite pixel format changed to BGRA for future use with native Windows drawing
- memory mapped files using is moved out, game files are not so big for this
- we may create another data sources (Amiga, other) and define priority in data.cc

### CODE REVIEW

File: src/popup.cc [hunk 1]
```diff
@@ -1795,18 +1795,18 @@ popup_box_t::draw_knight_level_box() {
   draw_box_background(311);
 
   player_t *player = interface->get_player();
-  draw_green_string(8, 8, level_str[player->knight_occupation[3] & 0xf]);
+  draw_green_string(8, 8, level_str[player->knight_occupation[3] & 0x8]);
```
Comment (jonls) [inline src/popup.cc]: Why this change?

File: src/popup.cc [hunk 2]
```diff
@@ -2099,7 +2099,7 @@ popup_box_t::draw_ordered_building_box() {
 
   int sprite = map_building_sprite[type];
   int x = 6;
-  if (sprite == 0xc0 /*stock*/ || sprite >= 0x9e /*tower*/) x = 4;
+  if (sprite == 0xc0 /*stock*/ || sprite < 0x9e /*tower*/) x = 4;
```
Comment (jonls) [inline src/popup.cc]: Why this change?

File: src/data-source-dos.cc [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+      }
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/* Load data file at path and let the global variable sprites refer to the memory
+ with the data file content. */
```
Comment (jonls) [inline src/data-source-dos.cc]: This comment seems to be inaccurate now as there is no global variable `sprites`.

File: src/data.cc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    res_path += suffix;
+  }
+
+  search_paths.push_back(res_path);
+}
+
+#define MAX_DATA_PATH      1024
+
+/* Load data file from path is non-NULL, otherwise search in
+ various standard paths. */
```
Comment (jonls) [inline src/data.cc]: This comment does not seem to be accurate anymore as the path cannot be NULL.

**Comment by jonls** [comment]
Did you manage to fix the issues that @zdechlak brought up?

File: src/popup.cc (hunk 1 above)
Comment (wdigger) [inline src/popup.cc]: level_str have only 5 values, chance to overboard lower

File: src/popup.cc (hunk 2 above)
Comment (wdigger) [inline src/popup.cc]: c0 always >= 9e, it's strange and I check with original code...

File: src/popup.cc (hunk 1 above)
Comment (wdigger) [inline src/popup.cc]: Of course 7! Shit on me!

File: src/data.cc (hunk 4 above)
Comment (wdigger) [inline src/data.cc]: @jonls window have no debug version of runtime libs by default, I advised him to download release binary.

**Comment by wdigger** [comment]
@jonls window have no debug version of runtime libs by default, I advised him to download release binary.

**Comment by wdigger** [comment]
Any further comments before merge?

**Comment by jonls** [comment]
I don't have any other comments :+1:
